test/e2e/operator_test.go

```diff
@@ -455,118 +280,76 @@ u3YLAbyW/lHhOCiZu2iAI8AbmXem9lW6Tr7p/97s0w==
+	if err := kclient.Update(context.TODO(), ic); err != nil {
 		t.Fatalf("failed to update ingresscontroller: %v", err)
 	}
 
 	// Wait for the deployment scale up to be observed.
-	err = waitForAvailableReplicas(cl, name, 2*time.Minute, newReplicas)
-	if err != nil {
-		t.Fatalf("failed waiting deployment %s to scale to %d: %v", deploymentName, newReplicas, err)
+	if err := waitForAvailableReplicas(kclient, ic, 2*time.Minute, newReplicas); err != nil {
+		t.Fatalf("failed waiting deployment %s to scale to %d: %v", defaultName, newReplicas, err)
 	}
 
 	// Ensure the ingresscontroller remains available
-	err = waitForIngressControllerCondition(cl, 2*time.Minute, name, defaultAvailableConditions...)
-	if err != nil {
-		t.Errorf("failed to observe expected conditions: %v", err)
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
 	}
 
 	// Scale back down.
-	if err := cl.Get(context.TODO(), name, ic); err != nil {
-		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	if err := kclient.Get(context.TODO(), defaultName, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller: %v", err)
 	}
-	ic.Spec.Replicas = &originalReplicas
-	if err := cl.Update(context.TODO(), ic); err != nil {
+	ic.Spec.Replicas = originalReplicas
+	if err := kclient.Update(context.TODO(), ic); err != nil {
 		t.Fatalf("failed to update ingresscontroller: %v", err)
 	}
 
 	// Wait for the deployment scale down to be observed.
-	err = waitForAvailableReplicas(cl, name, 2*time.Minute, originalReplicas)
-	if err != nil {
-		t.Fatalf("failed waiting deployment %s to scale to %d: %v", deploymentName, originalReplicas, err)
+	if err := waitForAvailableReplicas(kclient, ic, 2*time.Minute, *originalReplicas); err != nil {
+		t.Fatalf("failed waiting deployment %s to scale to %d: %v", defaultName, originalReplicas, err)
 	}
 
 	// Ensure the ingresscontroller remains available
-	err = waitForIngressControllerCondition(cl, 2*time.Minute, name, defaultAvailableConditions...)
-	if err != nil {
-		t.Errorf("failed to observe expected conditions: %v", err)
-	}
-}
-
-func waitForAvailableReplicas(cl client.Client, name types.NamespacedName, timeout time.Duration, expectedReplicas int32) error {
-	ic := &operatorv1.IngressController{}
-	var lastObservedReplicas int32
-	err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
-		if err := cl.Get(context.TODO(), name, ic); err != nil {
-			return false, nil
-		}
-		lastObservedReplicas = ic.Status.AvailableReplicas
-		if lastObservedReplicas != expectedReplicas {
-			return false, nil
-		}
-		return true, nil
-	})
-	if err != nil {
-		return fmt.Errorf("failed to achieve expected replicas: last observed : %v", lastObservedReplicas)
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
```

Comment:
Is `waitForIngressControllerCondition` really what we want, or do we want to monitor for 5 minutes to ensure conditions hold steady, as the comment suggests?  (Good candidate for a follow-up.)

Reply:
Added a TODO